infra/rush-x/src/actions/publish/packages.ts

```
@@ -0,0 +1,73 @@
+export const validateAndGetPackages = (options: PublishOptions) => {
+  const retrivePatterns = Object.values(RetrivePattern);
+  if (retrivePatterns.every(pattern => options[pattern]?.length <= 0)) {
+    throw new Error('No packages to publish');
+  }
+  return retrivePatterns.reduce((acc, pattern) => {
+    const packages = options[pattern];
+    if (!packages || packages.length <= 0) {
+      return acc;
+    }
+    const placeholders = retrivePackages(pattern as RetrivePattern, packages);
+    return new Set([...acc, ...placeholders]);
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)